### ISSUE TITLE
Hex diff - Hex compare 

### ISSUE THREAD

**Issue by mbyt** [microsoft/vscode-hexeditor#47]
Thank you for this marvelous extension. Editing and viewing binary files is quite cumbersome. This extension definitely helps here!

Especially when binary files were changed, it is really handy to compare them in HEX. That way I can easily check that I didn't destroy anything. Therefore the question: Is it possible to add support for comparison / compare editors in HEX?

I am only aware of the Total Commander being capable to do that (_all I want to say is this is a really helpful feature, which not many tools out there support_). Unfortunately I do not speak Java- nor Typescript...

**Comment by Pirulax** [comment]
Well, this is kinda disappointing.

**Comment by drognisep** [comment]
Yeah, this would be really nice to have.

**Comment by lramos15** [comment]
I can look at what total commander does. This problem has many cases of what the comparison would do. Would it be Byte by byte, then what happens if a singular byte is added in the middle? Is the rest a mismatch? Do we try to find the largest match and then offset the two files to start at that largest match and then do a compare. I think the problem is a complicated one but definitely beneficial. I will look into it.

**Comment by drognisep** [comment]
Agreed, definitely complicated. It's not as simple as a Git diff, but a similar approach could probably be used. Git works on a per-line basis and defines chunks of similarity based on line boundaries and difference distance. Examining that procedure could probably inspire a similar approach here.

**Comment by Trass3r** [comment]
For reference: https://www.sweetscape.com/010editor/manual/Compare.htm

**Comment by stef-levesque** [comment]
TBH, since Microsoft is now making its own [Hex Editor](/microsoft/vscode-hexeditor), I wonder if my extension is still relevant.

**Comment by szszoke** [comment]
My current use-case where this extension would have been awesome was about comparing binary files with the same length. I was just curios about which bytes were changed and at which offset.

**Comment by pnvnd** [comment]
This would be an awesome feature!  I agree, comparing binary files of the same length and highlight where the differences are.  I currently use [WinMerge](https://winmerge.org/) to find differences, then back to a Hex editor to view what the the actual changes are.

**Comment by NQNStudios** [comment]
If #345 was marked completed, could this feature now be added?

**Comment by lramos15** [comment]
#345 isn't complete. it was just a duplicate of this one

**Comment by sabre230** [comment]
This is weirdly lacking still, comparing two binaries in hex view really needs to highlight those differences. This would be a giant boost in productivity for many instead of having to work through an array of different programs to perform the same tasks.

**Comment by docteurzoidberg** [comment]
+1 !

**Comment by hongshui3000** [comment]
+1 !

**Comment by DraTeots** [comment]
Please!

**Comment by keremenci** [comment]
+1

**Comment by bzboi** [comment]
+1

**Comment by CptHolzschnauz** [comment]
Agreed. Important feature

**Comment by deanqx** [comment]
When?

**Comment by whatsupbros** [comment]
+1 this is a really needed feature to i.e. compare raw serialized records (Avro, Protobuf, etc..) Thanks!

**Comment by esba1ley** [comment]
Agreed.  I'd use this if it were part of my VS Code tools.  I also agree that [Beyond Compare](https://www.scootersoftware.com/v4help/index.html?viewhex.html) is an excellent goal to shoot for.

**Comment by Dom324** [comment]
+9999
The hexeditor plugin is really nice, but it is really missing some diff/compare capability.

**Comment by Dom324** [comment]
Btw this issue is n˚1 when you sort either by number of comment, or thumbs up reactions. So I guess it is fair to say this is the most wanted feature.
![image](https://github.com/microsoft/vscode-hexeditor/assets/57506340/740f1906-1415-4854-a8f6-85d67684518e)

**Comment by AjaniBilby** [comment]
Diagnosing wasm binaries without this is a real pain

**Comment by AjaniBilby** [comment]
Update: TIL about [biodiff](https://github.com/8051Enthusiast/biodiff) which does basically what most people are asking for but in CLI, and it's pretty platform agnostic as upposed to other solutions.

**Comment by 1two3code** [comment]
I hacked together a tool (`0xdiff`) as a makeshift substitute until this feature is implemented in the vscode-hexeditor extension.
It's executed from the CLI but runs in the default _text_ diff editor in vscode by starting `code --wait --diff`.

It's a 0 dependency script with a few useful options that approximate some of the features in vscode-hexeditor.

### [Check it out here 🤙](https://github.com/1two3code/0xdiff)

Please feel free to hack and contribute if you have ideas, find bugs or feel like
there are some crucial features missing.

![](https://raw.githubusercontent.com/1two3code/0xdiff/main/docs/debug_Float.png)

#### tl;dr
`npm i -g 0xdiff`  
`0xdiff --help`

Hope it's useful 🙏

**Comment by sj0000e** [comment]
If you need basic functionality, try my plugin.
https://marketplace.visualstudio.com/items?itemName=sj0000e.hexdiff

**Comment by CptHolzschnauz** [comment]
> If you need basic functionality, try my plugin. https://marketplace.visualstudio.com/items?itemName=sj0000e.hexdiff

Looks cool, but how to? If i open two bin's in MS hex editor, it's not offering that dialog? Which hex editor is the right one?

May i suggest a small step by step instruction on the github repo home with all prerequisites needed for dummies like me?
Thx anyway

**Comment by hongshui3000** [comment]
> If you need basic functionality, try my plugin. https://marketplace.visualstudio.com/items?itemName=sj0000e.hexdiff

Great extension, most sufficient for me

**Comment by tomilho** [comment]
I'll try to implement this.

**Comment by Askhento** [comment]
bump!
